### ISSUE TITLE
tests that compare stdout and stderr with a reference may fail

### ISSUE THREAD

**Issue by mrdudz** [cc65/cc65#1912]
Some tests pipe the stdout and stderr of the compiler or assembler into a file and compare the result with a reference. This has the subtle problem that both stdout and stderr and buffered i/o streams that may or may not work the same on different OSs or even shells. That means the order of the lines in the resulting file is not guaranteed.

In practise it is "mostly" not an issue, but "we" still stumbled about one test where different behaviour can be triggered depending on running it in cmd.exe or bash (msys): ```test/asm/listing``` output of ```010-paramcount.bin``` differs from the reference output when running the test from cmd.exe.

The solution is most likely to have two reference files and not redirect into one file.

Fixing all the tests seems like a huge effort for little gain right now - but fixing enough that at least this one failing one works again on both would be nice - @spiro-trikaliotis perhaps you can have a look?

**Comment by spiro-trikaliotis** [comment]
One question: As I do not have a cmd.exe environment, can I have the output files (testwrk/asm/listing/) of a run with cmd.exe, so I can determine what fails exactly?

**Comment by mrdudz** [comment]
sure, with cmd.exe 
```
asm\listing>type ..\..\..\testwrk\asm\listing\010-paramcount.err
 .paramcount = 3
 .paramcount = 5
 .paramcount = 3
 .paramcount = 5
 .paramcount = 1
 .paramcount = 5
010-paramcount.s:18: Warning: User warning: r1 is blank!
010-paramcount.s:14: Note: Macro was defined here
010-paramcount.s:8: Note: Macro was defined here
010-paramcount.s:19: Warning: User warning: r1 is blank!
010-paramcount.s:14: Note: Macro was defined here
010-paramcount.s:8: Note: Macro was defined here
010-paramcount.s:20: Warning: User warning: r1 is blank!
010-paramcount.s:14: Note: Macro was defined here
010-paramcount.s:8: Note: Macro was defined here
```
hower the reference file (which apparently matches the output in msys or linux):
```
test\asm\listing>type ref\010-paramcount.err-ref
 .paramcount = 3
 .paramcount = 5
010-paramcount.s:18: Warning: User warning: r1 is blank!
010-paramcount.s:14: Note: Macro was defined here
010-paramcount.s:8: Note: Macro was defined here
 .paramcount = 3
 .paramcount = 5
010-paramcount.s:19: Warning: User warning: r1 is blank!
010-paramcount.s:14: Note: Macro was defined here
010-paramcount.s:8: Note: Macro was defined here
 .paramcount = 1
 .paramcount = 5
010-paramcount.s:20: Warning: User warning: r1 is blank!
010-paramcount.s:14: Note: Macro was defined here
010-paramcount.s:8: Note: Macro was defined here
```

**Comment by mrdudz** [comment]
should be solved by #1916
